ElementX/Sources/Services/Notification/Manager/NotificationManager.swift

```diff
@@ -128,6 +128,27 @@ final class NotificationManager: NSObject, NotificationManagerProtocol {
             .map(\.request.identifier)
         notificationCenter.removeDeliveredNotifications(withIdentifiers: notificationsIdentifiers)
     }
+    
+    func removeDeliveredNotificationsForFullyReadRooms(_ rooms: [RoomSummary]) async {
+        let roomsToLastMessageDates = rooms
+            .filter { $0.hasUnreadMessages == false }
+            .reduce(into: [:]) { partialResult, roomSummary in
+                partialResult[roomSummary.id] = roomSummary.lastMessageDate
+            }
+        
+        let notificationsIdentifiers = await notificationCenter
+            .deliveredNotifications()
+            .filter {
+                if let roomID = $0.request.content.roomID, let date = roomsToLastMessageDates[roomID] {
+                    $0.date <= date
```

Comment:
Just to help a bit with readability
```suggestion
            .filter { notification in
                if let roomID = notification.request.content.roomID, let lastMessageDate = roomsToLastMessageDates[roomID] {
                    notification.date <= lastMessageDate
```
Would be even nicer if it was a `guard let` too :)